ui/src/lib/utils/format-permission-mode.util.ts

```diff
@@ -0,0 +1,23 @@
+import { S_IFDIR, S_IFLNK, S_IFMT } from '@/constants/file-bit-masks'
+
+export const formatPermissionMode = (value: number): string => {
+  const permissionSymbols = ['x', 'w', 'r']
+
+  let result = ''
+
+  for (let i = 0; i < permissionSymbols.length; i++) {
+    for (let j = 0; j < permissionSymbols.length; j++) {
+      result = ((value >> (i * permissionSymbols.length + j)) & 1) !== 0 ? permissionSymbols[j] + result : '-' + result
+    }
+  }
```

Comment:
looks complicated
Why not
```suggestion
  const permissionSymbols = ['x','w','r','x','w','r','x','w','r']

  let result = ''.join(permissionSymbols.map((v, ind) => {
      if ((value >> (ind)) & 1) {
          return v;
      } else {
          return '-';
      }
  })
```